app/javascript/components/conditions/SkipLogicFormField/component.js

```diff
@@ -48,10 +68,38 @@ class SkipLogicFormField extends React.Component {
         <select {...selectProps}>
           {this.skipOptionTags()}
         </select>
-        <SkipRuleSetFormField
-          hide={skip === 'dont_skip'}
-          {...this.props}
-        />
+        <div
+          className="rule-set"
+          style={{ display: skip === 'dont_skip' ? 'none' : '' }}
+        >
+          {skipRules.map((rule, index) => (
+            <SkipRuleFormField
+              formId={formId}
+              hide={skip === 'dont_skip'}
+              key={rule.key || rule.id}
+              laterItems={laterItems}
+              ruleId={`rule-${index + 1}`}
+              namePrefix={`${type}[skip_rules_attributes][${index}]`}
+              refableQings={refableQings}
+              {...rule}
+            />
+          ))}
+          <div
+            className="rule-add-link-wrapper"
+          >
+            {/* TODO: Improve a11y. */}
+            {/* eslint-disable */}
+            <a
+              onClick={this.handleAddClick}
+              tabIndex="0"
+            >
+            {/* eslint-enable */}
+              <i className="fa fa-plus" />
+              {' '}
+              {I18n.t('form_item.add_rule')}
+            </a>
+          </div>
+        </div>
```

Comment:
Here I combined SkipRuleSet and SkipLogic because there didn't seem to be a need for both and the extra layer of nesting was increasing complexity.